File: app/src/main/java/com/sd/myandroid/page_holder/MainActivity.java
package com.sd.myandroid.page_holder;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.sd.lib.holder.page.FPageHolder;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        test();
    }

    private void test() {
        final FPageHolder holder = new FPageHolder();

        assert holder.getCurrentPage() == 0;
        assert holder.getPageForRefresh() == 1;
        assert holder.getPageForLoadMore() == 1;
        assert holder.getPageForRequest(false) == 1;
        assert holder.getPageForRequest(true) == 1;
        assert !holder.hasNextPage();

        holder.onSuccess(false)
                .setHasData(true)
                .setHasNextPage(true)
                .update();

        assert holder.getCurrentPage() == 1;
        assert holder.getPageForRefresh() == 1;
        assert holder.getPageForLoadMore() == 2;
        assert holder.getPageForRequest(false) == 1;
        assert holder.getPageForRequest(true) == 2;
        assert holder.hasNextPage();
    }
}

File: app/src/main/java/com/sd/demo/page_holder/MainActivity.kt
package com.sd.demo.page_holder

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.sd.demo.page_holder.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private val _binding by lazy { ActivityMainBinding.inflate(layoutInflater) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(_binding.root)
    }
}

inline fun logMsg(block: () -> String) {
    Log.i("page-holder-demo", block())
}